rpcserver.go

```diff
@@ -2130,13 +2130,41 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 		options = append(options, zpay32.CLTVExpiry(uint64(defaultDelta)))
 	}
 
+	// If some additional routing info is set, we'll make sure to include it
+	// in the invoice as well.
+	if len(invoice.RoutingInfo) > 0 {
```

Comment:
Rather than have the user manually populate the routing info (which seems like quite a burden), we should instead automatically scan our set of private channels, and create routing hints as necessary. This logic would need to traverse the graph one hop from our private channel, to find the _sets_ of channels that the sender can use to successfully route over the last mile of our private channel.

Reply:
Fixed.